src/plugins/workspace/public/components/workspace_list/index.tsx

```diff
@@ -3,45 +3,50 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useState } from 'react';
+import React, { useState, useMemo, useCallback } from 'react';
 import {
   EuiPage,
   EuiPageBody,
   EuiPageHeader,
   EuiPageContent,
-  EuiBasicTable,
   EuiLink,
-  Direction,
-  CriteriaWithPagination,
+  EuiButton,
+  EuiInMemoryTable,
+  EuiTableSelectionType,
+  EuiSearchBarProps,
 } from '@elastic/eui';
 import useObservable from 'react-use/lib/useObservable';
-import { useMemo, useCallback } from 'react';
 import { of } from 'rxjs';
 import { WorkspaceAttribute } from '../../../../../core/public';
 
 import { useOpenSearchDashboards } from '../../../../../plugins/opensearch_dashboards_react/public';
-import { switchWorkspace } from '../utils/workspace';
+import { switchWorkspace, updateWorkspace } from '../utils/workspace';
+import { debounce } from '../utils/common';
+
+import { formatUrlWithWorkspaceId } from '../../utils';
+
+import { WORKSPACE_CREATE_APP_ID } from '../../../common/constants';
+
+const WORKSPACE_LIST_PAGE_DESCRIPTIOIN =
+  'Workspace allow you to save and organize library items, such as index patterns, visualizations, dashboards, saved searches, and share them with other OpenSearch Dashboards users. You can control which features are visible in each workspace, and which users and groups have read and write access to the library items in the workspace.';
 
 export const WorkspaceList = () => {
   const {
     services: { workspaces, application, http },
   } = useOpenSearchDashboards();
 
-  const [pageIndex, setPageIndex] = useState(0);
-  const [pageSize, setPageSize] = useState(5);
-  const [sortField, setSortField] = useState<'name' | 'id'>('name');
-  const [sortDirection, setSortDirection] = useState<Direction>('asc');
-
+  const initialSortField = 'name';
+  const initialSortDirection = 'asc';
   const workspaceList = useObservable(workspaces?.workspaceList$ ?? of([]), []);
+  const [queryInput, setQueryInput] = useState<string>('');
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    pageSize: 5,
+    pageSizeOptions: [5, 10, 20],
+  });
 
-  const pageOfItems = useMemo(() => {
-    return workspaceList
-      .sort((a, b) => {
-        const compare = a[sortField].localeCompare(b[sortField]);
-        return sortDirection === 'asc' ? compare : -compare;
-      })
-      .slice(pageIndex * pageSize, (pageIndex + 1) * pageSize);
-  }, [workspaceList, pageIndex, pageSize, sortField, sortDirection]);
+  // Will be uesed when updating table actions
+  const [_, setSelection] = useState<WorkspaceAttribute[]>([]);
```

Comment:
Is that means this state won't be used? How about assign an empty function?

Reply:
This state will be used when further mock up update.